esp/logging/logginglib/LogFailSafe.cpp

```diff
@@ -83,16 +86,16 @@ CLogFailSafe::~CLogFailSafe()
 
 void CLogFailSafe::readCfg(IPropertyTree* cfg)
 {
-    StringBuffer safeRolloverThreshold(cfg->queryProp(PropSafeRolloverThreshold));
+    StringBuffer safeRolloverThreshold(queryProp(cfg, PropSafeRolloverThreshold));
```

Comment:
I was given a requirement for new code to support a configuration defined using yaml markup. The yaml equivalent for an element containing text content instead of attributes is deemed unacceptable, meaning that all so-called integration properties presently defined as elements with content must be readable as element attributes.

The queryProp and getProp functions enable yaml-compatible markup to be read by the manager, thread, failsafe, et al,  while still supporting existing xml configurations. When asked for an element value, it first checks for an attribute value with the same name because attribute values will be the preferred form going forward. The check for a slash is a sanity check just in case the function is called with a more complex xpath.

Reply:
Now it makes sense.